store/src/java/com/zimbra/cs/filter/SieveVisitor.java

```diff
@@ -284,14 +292,14 @@ private void acceptTest(Node node, RuleProperties props) throws ServiceException
         } else {
             if ("allof".equalsIgnoreCase(nodeName)) {
                 props.condition = Sieve.Condition.allof;
-                visitRule(node, VisitPhase.begin, props);
+                visitAllOfTest(node, VisitPhase.begin, props);
                 accept(node, props);
-                visitRule(node, VisitPhase.end, props);
+                visitAllOfTest(node, VisitPhase.end, props);
```

Comment:
Please see the fix description.   Source-code-wise, the method name explains its functionality .  Besides the the declaration of the visitAllOfTest() and visitAnyOfTest() have the same format as other visit*Test() methods.